features/storage/kvstore/DirectAccessDevicekey.cpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// ----------------------------------------------------------- Includes -----------------------------------------------------------
+
+#include "DirectAccessDevicekey.h"
+#include <string.h>
+#include <stdio.h>
+#include "mbed_error.h"
+#include "MbedCRC.h"
+#include "mbed_trace.h"
+#define TRACE_GROUP "DADK"
+
+using namespace mbed;
+
+// --------------------------------------------------------- Definitions ----------------------------------------------------------
+#define TDBSTORE_NUMBER_OF_AREAS 2
+#define MAX_DEVICEKEY_DATA_SIZE 64
+#define RESERVED_AREA_SIZE (MAX_DEVICEKEY_DATA_SIZE+8) /* DeviceKey Max Data size + 4bytes address + 4bytes deviceKey actual size */
```

Comment:
a `sizeof(reserved_trailer_t)` is cleaner than 8 here.

Reply:
changing..